src/scripts/main.js

```diff
@@ -1,3 +1,289 @@
+
+  const descriptionNot = document.createElement('p');
+
+  descriptionNot.textContent = description;
+
+  fragmentNot.append(titleNot);
+  fragmentNot.append(descriptionNot);
+  notification.append(fragmentNot);
+
+  document.body.append(notification);
+
+  setTimeout(() => {
+    notification.hidden = true;
+  }, 2000);
+};
+
+formBtn.addEventListener('click', (e) => {
+  e.preventDefault();
+
+  const formValues = {
+    employeeName: form.elements.name.value,
+    position: form.elements.position.value,
+    office: form.elements.office.value,
+    age: form.elements.age.value,
+    salary: form.elements.salary.value.trim(),
+  };
+
+  const formSalaryValue =
+    !formValues.salary || isNaN(formValues.salary)
+      ? ''
+      : '$' + addSeparator(+formValues.salary);
+
+  const fields = validateFormFields(formValues);
+
+  pushNotification(fields[0]);
+
+  if (fields[0].type === 'success') {
+    const row = addRow([
+      formValues.employeeName,
+      formValues.position,
+      formValues.office,
+      formValues.age,
+      formSalaryValue,
+    ]);
+
+    tbody.append(row);
+  }
+});
+
+function validateFormFields({ employeeName, position, office, age, salary }) {
+  const notifications = [];
+
+  if (employeeName.length < 4) {
+    notifications.push(
+      new CustomNotification(
+        'Error',
+        'Name must be at least 4 characters long',
+        'error',
+      ),
+    );
+  }
+
+  if (!position.trim()) {
+    notifications.push(
+      new CustomNotification('Error', 'Position cannot be empty', 'error'),
+    );
+  }
+
+  if (!office.trim() || !offices.includes(office)) {
+    notifications.push(
+      new CustomNotification('Error', 'Office cannot be empty', 'error'),
+    );
+  }
+
+  if (age < 18 || age > 90) {
+    notifications.push(
+      new CustomNotification('Error', 'Age must be between 18 and 90', 'error'),
+    );
+  }
+
+  if (salary <= 0) {
+    notifications.push(
+      new CustomNotification(
+        'Error',
+        'The value of salary must be a number greater than 0',
+        'error',
+      ),
+    );
+  }
+
+  if (notifications.length === 0) {
+    notifications.push(
+      new CustomNotification('Success', 'Data successfully added', 'success'),
+    );
+  }
+
+  return notifications;
+}
+
+function addRow(fields) {
+  const row = document.createElement('tr');
+
+  fields.forEach((field) => {
+    const td = document.createElement('td');
+
+    td.textContent = field;
+    row.append(td);
+  });
+
+  return row;
+}
+
+const headers = document.querySelectorAll('thead th');
+
+headers.forEach((header) => {
+  header.addEventListener('click', () => {
+    const index = header.cellIndex;
+    const currentSort = header.dataset.sort;
+    const newSortOrder =
+      !currentSort || currentSort === 'DESC' ? 'ASC' : 'DESC';
+
+    headers.forEach((h) => delete h.dataset.sort);
+    header.dataset.sort = newSortOrder;
+
+    let rows = [...table.querySelectorAll('tr:has(td')];
```

Comment:
The selector 'tr:has(td' seems to be incorrect or incomplete. Ensure that the selector is properly closed and valid for the intended use.

Reply:
"If the AI previously approved the changes, but some inaccuracies were still noted, I made the necessary corrections and submitted a new pull request. However, the review didn't take place; instead, the new commit was immediately marked as 'rejected' without any explanation. Should I expect a review, or does the AI-mentor stop responding to requests after the initial approval, and further assistance is no longer available?"